### PR TITLE
Hints Simplification

### PR DESCRIPTION
**Pull request by or-ram**
I'm excited to present this pull request, which introduces a significant improvement to our codebase by allowing the inclusion of hints in JavaScript (JS) objects. This change aims to simplify our current approach and streamline the overall workflow.

Here's a brief overview of the changes included in this pull request:

New Hint Feature: 
I've added a new property, which allows for the easy pass of type hints. This will make it easier for developers to FastRPC.

Code Simplification: By incorporating hints into the objects, we're able to reduce the complexity of generating hints, making code more readable and maintainable. This could benefit all as it will be easier to understand the code and collaborate on it.

I kindly request all members to thoroughly review the changes in this pull request and provide any feedback or suggestions you may have. Your input is valuable and crucial to ensuring the success of this new feature.

Thank you for your cooperation and continued dedication to improving our codebase. I look forward to hearing your thoughts and working together to make this enhancement a reality.

Best regards,
Roman

### CODE REVIEW

File: typescript/fastrpc.ts [hunk 1]
```diff
@@ -303,8 +317,10 @@ function _serializeValue(result: BYTES, value: any) {
 				_serializeArrayBuffer(result, value);
 			} else if (value instanceof Date) {
 				_serializeDate(result, value);
-			} else if (value instanceof Array) {
+			} else if (isBinary(value) || value instanceof Array) {
 				_serializeArray(result, value);
+			} else if (isDouble(value)) {
+				_serializeDouble(result, value);
```
Comment (ondras) [inline typescript/fastrpc.ts:323]: Nebylo by lepsi tady psat `_serializeDouble(result, value.value)`, cimz by pak v implementaci `_serializeDouble` nemusel byt na zacatku ten test?

File: typescript/fastrpc.ts [hunk 2]
```diff
@@ -327,8 +343,18 @@ function _serializeArrayBuffer(result: BYTES, data: ArrayBuffer) {
 	return;
 }
 
-function _serializeArray(result: BYTES, data: any[]) {
-	if (_getHint() == "binary") {
+function _serializeArray(result: BYTES, data: any[] | Binary) {
```
Comment (ondras) [inline typescript/fastrpc.ts:346]: Dva komentare:
  - neni implementace pro varianty `isBinary` a `_getHint() == "binary"`  stejna? Pokud ano, nemusela by byt ve dvou ruznych vetvich...
  - neni nahodou ten vypocet prvnich par bajtu spolecny pro vsechny tri vetve? Pokud ano, dal by se udelat spolecne (a lisila by se pak jen serializace samotneho pole, bud pres `_append` cele promenne, nebo cyklem pro `ArrayBuffer`...

File: typescript/fastrpc.ts (hunk 1 above)
Comment (ondras) [reply typescript/fastrpc.ts:323]: Prijde mi vzdycky lepsi, kdyz funkce (v tomhle pripade `_serializeDouble`) nemusi umet zpracovavat parametry ruznych datovych typu.

Comment (or-ram) [reply typescript/fastrpc.ts]: Vyřešeno

File: typescript/fastrpc.ts (hunk 2 above)
Comment (or-ram) [reply typescript/fastrpc.ts]: Podivej se jak jsem to zkusil vyřešit

Comment (ondras) [reply typescript/fastrpc.ts]: Nekam tam uplne zmizelo to `TYPE_ARRAY << 3;`, ne?

Asi ta funkce `_serializeArray` zacina byt prilis chytra: na vstupu prijima bud JS pole hodnot, nebo JS pole s hintem "binary", nebo JS objekt s `_hint:"binary"`. A na vystupu generuje bud TYPE_ARRAY nebo TYPE_BINARY.

Co kdybychom to teda rozdelili na dve? Jedna, ktera *vzdy* produkuje TYPE_ARRAY, a druha, ktera *vzdy* produkuje TYPE_BINARY (a do ni se posilaji arraybuffery a obe dve hintovane varianty).

Comment (or-ram) [reply typescript/fastrpc.ts]: Jo to dává smysl protože i tak ta funkce `_serializeArray` byla spíše sloučeninou dvou funkcí. takhle to bude přehlednější.

**Comment by or-ram** [comment]
> @or-ram paradni, ted to vypada dobre. Testy prochazi? (`cd test; node runner.mjs`)

Jo testy procházi.

**Pull request merged**